packages/astro/src/content/runtime.ts

```diff
@@ -319,9 +334,84 @@ type RenderResult = {
 	remarkPluginFrontmatter: Record<string, any>;
 };
 
+const IMAGE_REGEX = /__ASTRO_IMAGE_="([^"]+)"/g;
```

Comment:
nit: `CONTENT_LAYER_IMAGE_REGEX`

Mainly to give a more descriptive name, so we know it's not a "generic" image.